### DISCUSSION TITLE
python lsp works(kinda?) but no highlighting

### DISCUSSION THREAD

**Discussion by jrharting** [helix-editor/helix#5587] [Q&A]
python lsp kind of working, but tree-sitter fails to open the python.so library and I get code hints bt no highlighting.

system:
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"

Steps taken:
1. install pylsp via conda
2. install hx from source following docs
3. link runtime folder to ~/.config/helix/runtime
4. hx -g fetch && hx -g build

When I open a python file, it takes a few seconds but I do get code hints (see image), however, there is no highlighting.

Contents of log:
$ cat  ~/.cache/helix/helix.log
2023-01-18T17:42:56.726 helix_core::syntax [ERROR] Failed to load tree-sitter parser for language "python": Error opening dynamic library "/home/UNIXHOME/jharting/.config/helix/runtime/grammars/python.so"
2023-01-18T17:43:01.321 helix_term::application [ERROR] Timed out waiting for language servers to shutdown

The python.so file exists:
$ ll /home/UNIXHOME/jharting/.config/helix/runtime/grammars/python.so
-rwxr-xr-x 1 jharting Domain Users 561K Jan 18 17:47 /home/UNIXHOME/jharting/.config/helix/runtime/grammars/python.so

Expected: Syntax highlighting 

![Screen Shot 2023-01-18 at 6 03 44 PM](https://user-images.githubusercontent.com/3595985/213338323-e957d192-6da6-4650-9fde-e7698e4fb934.png)

Highlighting works fine in rust and the hx health for python looks fine:

$ hx --health python
Configured language server: pylsp
Binary for language server: /home/UNIXHOME/jharting/miniconda3/bin/pylsp
Configured debug adapter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓

Is there a tree-sitter log I can find somewhere to diagnose this? What else can I try?

**Comment by archseer** [comment]
> helix_core::syntax [ERROR] Failed to load tree-sitter parser for language "python": Error opening dynamic library "/home/UNIXHOME/jharting/.config/helix/runtime/grammars/python.so"

You might have to rebuild your grammars, I'd wipe `runtime/grammars/*.so` and `hx --grammar build`

**Reply by jrharting** [reply]
Thanks for the suggestion, but I have tried that multiple times to no avail.  I've uninstalled/cleared helix and reinstalled multiple times (like 5 times), removed/refetched/rebuilt grammars multiple times with different gcc versions, all without success, except once. 

The very first time I reinstalled helix everything seemed fine until I tried to edit the language.toml, after which python highlighting and sometimes code hints failed and I have not been able to make it work since, even after removing the language.toml I edited.

I just have no idea why it errors out -- pylsp and tree-sitter both appear to work from the command line, but I don't know how to diagnose that error above.  Is there a log for tree-sitter somewhere?

I'm to the point of abandoning helix for python until a newer version is released :(.

**Reply by archseer** [reply]
It seems specific to your system so I can't imagine a new version would fix anything. There's a few issues with the same error where users either had the wrong compiler set up, or had two clashing helix installations. https://github.com/helix-editor/helix/issues?q=is%3Aissue+Error+opening+dynamic+library+is%3Aclosed

**Reply by Diogenesoftoronto** [reply]
I also had this problem will try removing all the other builds of helix and I will try building it from source on main to see if I can reproduce the issue.
